web/src/stream_events.js

```diff
@@ -114,6 +117,13 @@ export function mark_subscribed(sub, subscribers, color) {
         return;
     }
 
+    if (sub.push_notifications_enabled) {
+        sub.push_notifications = sub.push_notifications_enabled;
+    } else {
+        sub.push_notifications = null;
```

Comment:
Can we instead just set `push_notifications` field to the `push_notifications` field in the object sent with the event.